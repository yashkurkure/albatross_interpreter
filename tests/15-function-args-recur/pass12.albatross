var mm int := 0;
var nn int := 0;

fun ackermann int (m int,n int) {
    if (m ==0 ) {
        return n+1;
    }
    if (m>0 && n==0) {
      return ackermann(m-1,1);
    }
    if (m>0 && n>0) {
      return ackermann(m-1,ackermann(m,n-1));
    }
    return 0;
}

while (mm <= 3) {
  nn := 0;
  while (nn <= 3) {
    printstring("m: ");
    printint(mm);
    printstring(" n: ");
    printint(nn);
    printstring(" => ");
    printint(ackermann(mm,nn));
    printstring("        ");

    nn := nn + 1;
  }

  mm := mm + 1;
}

return 0;