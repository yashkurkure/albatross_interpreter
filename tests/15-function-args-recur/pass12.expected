m: 0 n: 0 => 1        m: 0 n: 1 => 2        m: 0 n: 2 => 3        m: 0 n: 3 => 4        m: 1 n: 0 => 2        m: 1 n: 1 => 3        m: 1 n: 2 => 4        m: 1 n: 3 => 5        m: 2 n: 0 => 3        m: 2 n: 1 => 5        m: 2 n: 2 => 7        m: 2 n: 3 => 9        m: 3 n: 0 => 5        m: 3 n: 1 => 13        m: 3 n: 2 => 29        m: 3 n: 3 => 61        
